File: 2273-find-resultant-array-after-removing-anagrams/2273-find-resultant-array-after-removing-anagrams.py
class Solution:
    def removeAnagrams(self, words: List[str]) -> List[str]:
        ans = [words[0]]
        prev = "".join(sorted(words[0]))
        for i in range (1, len(words)):
            temp = "".join(sorted(words[i]))
            if (temp != prev) :
                ans.append(words[i])
                prev = temp
        return ans

File: 2273-find-resultant-array-after-removing-anagrams/2273-find-resultant-array-after-removing-anagrams.cpp
class Solution {
public:
    vector<string> removeAnagrams(vector<string>& words) {
        vector<string>ans;
        unordered_map<string, string>mp;
        for (const string& s: words){
            string temp = s;
            sort(temp.begin(), temp.end());
            mp[s] = temp;
        }
        for (int i = 0; i < words.size(); i++) {
            ans.push_back(words[i]);
            while (i + 1 < words.size() && mp[words[i]] == mp[words[i + 1]]) {
                i++;
            }
        }
        return ans;
    }
};